src/main.rs

```diff
@@ -38,6 +38,25 @@ fn main() {
         match tokens[0] {
             "args"                  => println!("args = {:?}", tokens),
             "exit" | "q" | "quit"   => break,
+            "add-node" => {
+                if tokens.len() < 2 || tokens.len() > 3 {
+                    println!("Syntax: add-node <home_id> [secure]");
+                    continue;
+                }
+                if let Ok(home_id) = u32::from_str_radix(tokens[1], 16) {
+                    let secure = tokens.len() >= 3 && tokens[2] == "secure";
```

Comment:
(may need a `str.to_owned()`, not entirely sure why)

Another option is `let secure = vec.get(2).map(|str| str == "secure").unwrap_or_default();`